test/src/missle_launcher_test.cpp

```diff
@@ -5,22 +5,68 @@
 #include "mocks/mock_missile_launcher_listener.h"
 
 using ::testing::_;
+using ::testing::Property;
+using ::testing::InSequence;
 
-class MissleLauncherTest : public ::testing::Test {
+class MissileLauncherTest : public ::testing::Test {
 protected:
+    MissileLauncher* launcher;
+    MockMissileLauncherListener* launcher_listener;
+
     virtual void SetUp() {
+        launcher = new MissileLauncher() ;
+        launcher_listener = new MockMissileLauncherListener();
+        launcher->add_listener(launcher_listener);
     }
 
     virtual void TearDown() {
+        delete launcher_listener;
```

Comment:
Not sure, google test will actually tell you if you have a memory leak. I only had to delete the listener to fix my memory leak error, deleting the actual launcher was not necessary. Not sure why but I think including unnecessary code is worse.